### PR TITLE
Migrate GoogleCloudStorageOptions to AutoValue

### PR DESCRIPTION
**Pull request by medb**

### CODE REVIEW

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java [hunk 1]
```diff
@@ -423,11 +423,12 @@ public ParentTimestampUpdateIncludePredicate(
...
      * Determine if updating directory timestamps should be ignored.
-     * @return True if the directory timestamp should not be updated. False to indicate it should
-     * be updated.
+     *
+     * @return True if the directory timestamp should not be updated. False to indicate it should be
+     *     updated.
      */
     @Override
-    public boolean shouldUpdateTimestamp(URI uri) {
+    public boolean test(URI uri) {
```
Comment (functicons) [inline gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java]: Why do we rename it? I feel "test" is a bad name, unclear what it actually means.

Comment (medb) [reply gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java]: This is a `test` method in `java.util.function.Predicate` - I just removed custom predicate class and switched to provided by JDK

**Comment by medb** [comment]
> Will this change break existing user code?

Yes, potentially it could break user's code, but for major version release (2.0.0) we don't care about this.

**Pull request merged**